pandas/tslib.pyx

```diff
@@ -3423,6 +3423,9 @@ def tz_convert(ndarray[int64_t] vals, object tz1, object tz2):
     trans, deltas, typ = _get_dst_info(tz2)
     trans_len = len(trans)
 
+    if (utc_dates==iNaT).all():
+        return utc_dates
+
```

Comment:
add this as an or condition to the lines below as its checking the same thing

e.g.

```
if (result==iNAT).all() or (utc_dates==iNaT).all():
    ....
```